server.js

```diff
@@ -15,19 +17,62 @@ import cors from 'cors'
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+const endPoints = require('express-list-endpoints')
+app.get("/", (request, response) => {
+  response.send(endPoints(app));
+});
+
+// Get all the books in the list & filter options
+app.get('/books', (req, res) => {
+  // get only books that have been ranked
+  const isRanked = req.query.isRanked
+
+  // get only books with a certain minimum ranking
+  const minRating = req.query.minRating;
+
+  // get only books by a certain author
+  const author = req.query.author;
+
+  let filteredBooks = books;
+
+  if (isRanked) {
+    filteredBooks = filteredBooks.filter((book) => book.ratings_count !== 0)
+  };
+
+  if (minRating) {
+    filteredBooks = filteredBooks.filter((book) => book.average_rating >= minRating)
+  };
+
+  if (author) {
+    filteredBooks = filteredBooks.filter((book) => book.authors.toLowerCase().includes(author.toLowerCase()))
+  };
+
+  res.json(filteredBooks);
+});
+
+// Get a specific book usig the ID
+app.get('/books/:id', (req, res) => {
+  const id = req.params.id;
+  const book = books.find((book) => book.bookID === +id);
+  res.json(book);
+});
+
+// Get all books in a specific language
+app.get('/languages/:language', (req, res) => {
+  const language = req.params.language;
+  const booksInLanguage = books.filter((book) => book.language_code === language);
+  res.json(booksInLanguage);
+});
```

Comment:
Would be nice if this too were a query under the `/books` endpoint, since it returns books and not languages ;)